Fakebook.Profile/Fakebook.Profile.DataAccess/EntityModel/EntityProfile.cs

```diff
@@ -1,50 +1,123 @@
+        /// <param name="profilePictureUrl">The url of the user's profile picture.</param>
+        public EntityProfile(string email, string firstName, string lastName, DateTime dob, Uri profilePictureUrl, string? phoneNumber = null, string? status = null, IList<string>? followerEmails = null, IList<string>? followingEmails = null)
+        {
+            Email = email;
+            FirstName = firstName;
+            LastName = lastName;
+            BirthDate = dob;
+            PhoneNumber = phoneNumber;
+            FollowerEmails = followerEmails;
+            FollowingEmails = followingEmails;
+            ProfilePictureUrl = profilePictureUrl;
+        }
+
+        /// <summary>
+        /// The user's ID.  The primary key.  Cannot be null.
         /// </summary>
         [Key]
-        [Column(name: nameof(Email))]
+        [Required]
+        public int UserId { get; set; }
+
+        /// <summary>
+        /// The user's unique email. Cannot be null.
+        /// </summary>
+        [Required]
+        [MaxLength(45),MinLength(5)]
         public string Email { get; set; }
 
+        /// <summary>
+        /// The user's followers. Can be null.
+        /// </summary>
+        [NotMapped]
+        public IList<string>? FollowerEmails { get; set; } 
+
+        /// <summary>
+        /// The user's followees. Can be null.
+        /// </summary>
+        [NotMapped]
+        public IList<string>? FollowingEmails { get; set; } 
+
         /// <summary>
         /// The Uri of the user's profile picture. Can be null.
         /// </summary>
-        public Uri ProfilePictureUrl { get; set; }
+        public Uri? ProfilePictureUrl { get; set; }
 
         /// <summary>
         /// The user's first name. Cannot be null.
         /// </summary>
+        [MaxLength(25)]
         public string FirstName { get; set; }
 
         /// <summary>
-        /// The user's last name, cannot be null.
+        /// The user's last name. Cannot be null.
         /// </summary>
+        [Required]
+        [MaxLength(25),MinLength(2)]
         public string LastName { get; set; }
 
         /// <summary>
-        /// The user's phone number. can be up to 15 characters. (?) Can be null.
+        /// The user's phone number. Can be 10 - 15 characters. Can be null.
         /// </summary>
-        [MaxLength(15)]
-        public string PhoneNumber { get; set; }
+        [MaxLength(15),MinLength(10)]
+        public string? PhoneNumber { get; set; }
 
         /// <summary>
-        /// The user's birthdate. Can't be null.
+        /// The user's birthdate. Cannot be null.
         /// </summary>
+        [Required]
         public DateTime BirthDate { get; set; }
 
         /// <summary>
-        /// The user's current status message, can be null.
+        /// The user's current status message. Can be null.
         /// </summary>
-        public string Status { get; set; }
+        [MaxLength(300)]
+        public string? Status { get; set; }
```

Comment:
string is a reference type so adding the `?` doesn't do anything, not unless we turn on nullable reference types feature which i don't think our main frameworks are fully ready for yet. i see it on PhoneNumber as well